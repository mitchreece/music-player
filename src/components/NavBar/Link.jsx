import React from 'react'
import { NavLink } from 'react-router-dom'
import PropTypes from 'prop-types'

const Link = (props) => {
	return (
		<li className="nav__links__link">
			<NavLink className="link"
				activeClassName="link--active"
				{...props.config}
			>
				{props.label}
			</NavLink>
		</li>
	)
}

Link.propTypes = {
	config: PropTypes.object.isRequired,
	label: PropTypes.string.isRequired,
}

export default Link
