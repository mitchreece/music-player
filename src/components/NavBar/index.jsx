import React from 'react'
import { nav } from '../../routes/routes.config'
import Link from './Link'

const NavBar = () => {
	return (
		<nav className="nav">
			<ul className="nav__links">
				{ 
					nav.map((link, i) => {
						return(
							<Link label={link.label} 
								config={link.config}
								key={i} /> 
						) 
					})
				}
			</ul>
		</nav>
	)
}

export default NavBar
