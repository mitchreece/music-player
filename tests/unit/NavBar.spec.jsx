import React from 'react'
import {MemoryRouter as Router} from 'react-router-dom'
import renderer from 'react-test-renderer'
import NavBar from '../../src/components/NavBar'

it('renders correctly', () => {
	const tree = renderer
		.create(<Router><NavBar /></Router>)
		.toJSON()

	expect(tree).toMatchSnapshot()
})
