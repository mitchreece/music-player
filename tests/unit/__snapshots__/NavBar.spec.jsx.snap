// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="nav"
>
  <ul
    className="nav__links"
  >
    <li
      className="nav__links__link"
    >
      <a
        aria-current="true"
        className="link link--active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
  </ul>
</nav>
`;
